.storybook/custom/components/DocsPage.js

```diff
@@ -0,0 +1,56 @@
+import '../custom.scss';
+import Community from './Community';
+import Footer from './Footer';
+import Header from './Header';
+import Import from './Import';
+import Toc from './Toc';
+import tocbot from 'tocbot';
+import React, { useEffect, useState } from 'react';
+import {
+    Title,
+    Subtitle,
+    Description,
+    Primary,
+    Props,
+    Stories,
+  } from '@storybook/addon-docs/blocks';
+
+const DocsPage = () => {
+
+    useEffect(() => {
+        tocbot.init(
+            {
+                tocSelector: '.js-toc',
+                contentSelector: '.sbdocs-wrapper',
+                headingSelector: 'h2.sbdocs-h2, h3.sbdocs-h3, h4.sbdocs-h4',
+                orderedList: true,
+                scrollContainer: '.sbdocs-wrapper',
+                collapseDepth: 3
+            }
+        );
+        document.querySelectorAll('.toc-link').forEach( x => x.setAttribute('target','_self'));
```

Comment:
This is ugly - can anyone think of another way to do this? It doesn't look like the tocbot api allows adding additional attributes. 

It's also not updating the active class (blue left border) as you scroll

Reply:
fixed